### ISSUE TITLE
Cast is redundant (IDE0004) analyzer bug

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn#6142]
There is a bug at IDE0004 Analyzer.

In following code snippet it proposes to remove 'enum to int' cast even though it changes runtime behaviour.

```c#
class Program
{
    static void Main()
    {
        Console.WriteLine($"{(int)MyEnum.EnumValue}");  //writes '0'
        Console.WriteLine($"{MyEnum.EnumValue}");       //writes 'EnumValue'
    }

    enum MyEnum
    {
        EnumValue
    }
}
```

**Comment by Pilchie** [comment]
I believe this is a dupe of something you already fixed for 1.1?

**Comment by DustinCampbell** [comment]
This is a dupe of https://github.com/dotnet/roslyn/issues/4037, which is already fixed for 1.1 with https://github.com/dotnet/roslyn/pull/5029.
